DbAccessor/src/Messages/CreateConversation.cs

```diff
@@ -0,0 +1,119 @@
+            try
+            {
+                convoRequest = JsonSerializer.Deserialize<ConversationRequest>(requestBody);
+
+                var (invalid, errorMessage) = convoRequest.validate();
+                if (invalid)
+                {
+                    return new BadRequestObjectResult(errorMessage);
+                }
+            }
+            catch (JsonException ex)
+            {
+                _logger.LogError($"JSON deserialization failed: {ex.Message}");
+                return new BadRequestObjectResult("Incomplete Conversation Request data.");
+            }
+            _logger.LogInformation($"Raw Request Body: {JsonSerializer.Serialize(requestBody)}");
+
+
+            // Get the user name and photo details by calling SQL Get user endpoint
+            User otherUser;
+            string endpoint = $"{_baseApiUrl}/api/GetUser";
+
+            // make http request using req.Headers and endpoint to get the user details
+            var (error, response) = await Utilities.MakeHttpGetRequest(convoRequest.UserId, endpoint);
+            if (!error && response != null)
+            {
+                otherUser = JsonSerializer.Deserialize<User>(response);
+                otherUser.UserId = convoRequest.UserId;
+            }else{
+                return new ObjectResult("Failed to get user details from DB") { StatusCode = StatusCodes.Status500InternalServerError };
+            }
+
+            Message newMessage = new Message
+            (
+                userId, // the user who initiated the conversation by sending a message is the "SenderId" of the message
+                convoRequest.Contents, 
+                convoRequest.TimeStamp
+            );
+            
+            // Create a document for DB insertion
+            var newConversation = new Conversation
+            (
+                new List<string> 
+                {
+                    convoRequest.UserId,
+                    userId
+                }, 
+                new List<Message> { newMessage }
+            );
+
+            // Response object includes User details, hence its seperate from the conversation object
+            var responseObj = new ConversationResponse
+            (
+                newConversation.ConversationId, 
+                otherUser, 
+                new List<Message> {newMessage} 
+            );
+
+            // Get the database collection and insert the new conversation
+            IMongoCollection<Conversation> convoCollection = client.GetDatabase("user_chats").GetCollection<Conversation>("conversations");
+
+            try
+            {
+                await convoCollection.InsertOneAsync(newConversation);
+                
+                return new OkObjectResult(responseObj);
+            }
+            catch (Exception ex)
+            {
+                return new ObjectResult($"Failed to insert document: {ex.Message}") { StatusCode = StatusCodes.Status500InternalServerError };
```

Comment:
More logging in places like this would be nice, but technically its fine since the Logic layer will likely log this error response itself